### PR TITLE
Strip path prefixes on profile frames

### PR DESCRIPTION
**Pull request by stayallive**
This was an open TODO that is resolved!

### CODE REVIEW

**Comment by ste93cry** [comment]
Wouldn't be better to make `PrefixStripper` a `final class` with a `static` method accepting the list of prefixes instead of using a `trait`?

**Comment by stayallive** [comment]
That could also work yes. I don't think there is a technical reason why either way is "better" so potato, potahto really in this case 😄

File: tests/Profiling/ProfileTest.php [hunk 1]
```diff
@@ -217,6 +218,176 @@ public static function formattedDataDataProvider(): \Generator
             ],
         ];
 
+        yield [
+            $event,
+            [
+                [
+                    'trace' => [
+                        [
+                            'file' => '/index.php',
```
Comment (cleptric) [inline tests/Profiling/ProfileTest.php:227]:
```suggestion
                            'file' => '/var/www/html/index.php',
```

and below, to actually test it 😬

**Review by cleptric — APPROVED** [review]
A quick test run with some app that reports to https://sentry.io would be appreciated, besides that, :shipit:

**Comment by ste93cry** [comment]
> That could also work yes. I don't think there is a technical reason why either way is "better"

Some of the reasons I can think of is that:

- Even if not impossible, they are more difficult to test than a `class` or a `method`
- They introduce horizontal inheritance, so understanding a code base can be significantly harder. Injecting a dependency is a lot cleaner in this aspect, as you make it clear what a class needs to work. But in this case, a `static` helper method would probably be enough
- Since a `trait` is basically a way to copy-paste some code in multiple places, you are effectively coupling it with the classes it will be used in, risking to make some change in one that doesn't suit the others. In this case the method does not access anything on $this, but you never know in the future what could happen.

Lately I've seen a lot of discussions around traits and why people think they are either good or bad. Honestly I don't have a real strong opinion on this, and in this case I agree that it doesn't really matter probably, but if it was me I would have created an helper class. Also, I don't like passing around all the client config when the function really needs just a plain `array` of strings.

If you really want to leave it as-is, I would at least add the `Trait` suffix to the trait to make it explicit what is it.

File: tests/Profiling/ProfileTest.php (hunk 1 above)
Comment (stayallive) [reply tests/Profiling/ProfileTest.php:227]: This one I don't get, I've duplicated the existing test case, to have one with and one without prefixes. Isn't that good enough for a test or did I misinterpret your comment?

(test case to sentry.io coming up!)

**Comment by stayallive** [comment]
For the requested sentry.io test (using a Laravel application):

[Before](https://devalex.sentry.io/profiling/profile/tny/e1d323e9f2984415b8633d8b17612418/flamechart/?colorCoding=by+system+vs+application+frame&query=&sorting=call+order&tid=0&view=top+down):

image

[After](https://devalex.sentry.io/profiling/profile/tny/2ec2959c321e496e8eaf1995701ed0b6/flamechart/?colorCoding=by+system+vs+application+frame&query=&sorting=call+order&tid=0&view=top+down):

image

Looks a lot better to me 😎

File: tests/Profiling/ProfileTest.php (hunk 1 above)
Comment (stayallive) [reply tests/Profiling/ProfileTest.php]: Oh I got it, 🤦 yes you are totally right... made the input the same for both test cases now 😅

**Pull request merged**